airflow/bin/cli.py

```diff
@@ -1003,16 +1004,33 @@ def connections(args):
     if args.add:
         # Check that the conn_id and conn_uri args were passed to the command:
         missing_args = list()
-        for arg in ['conn_id', 'conn_uri']:
+        invalid_args = list()
+        for arg in ['conn_id']:
             if getattr(args, arg) is None:
                 missing_args.append(arg)
+        if args.conn_uri:
+            for arg in ['conn_type', 'conn_host', 'conn_login', 'conn_password', 'conn_schema', 'conn_port']:
+                if getattr(args, arg) is not None:
+                    invalid_args.append(arg)
+        elif not args.conn_type:
+            missing_args.append('conn_uri or conn_type')
         if missing_args:
             msg = ('\n\tThe following args are required to add a connection:' +
                    ' {missing!r}\n'.format(missing=missing_args))
             print(msg)
+        if invalid_args:
+            msg = ('\n\tThe following args are not compatible with the ' +
+                   '--add flag and --conn_uri flag: {invalid!r}\n')
+            msg = msg.format(invalid=invalid_args)
+            print(msg)
+        if missing_args or invalid_args:
             return
 
-        new_conn = Connection(conn_id=args.conn_id, uri=args.conn_uri)
+        if args.conn_uri:
+            new_conn = Connection(conn_id=args.conn_id, uri=args.conn_uri)
+        else:
+            new_conn = Connection(conn_id=args.conn_id, conn_type=args.conn_type, host=args.conn_host,
```

Comment:
nits on indentation and space

Reply:
Is it wrong? I think it conforms to PEP8

Reply:
Sorry, my bad, mis-read the line wrap.